### DISCUSSION TITLE
How to find all items of a type?

### DISCUSSION THREAD

**Discussion by jduteau** [sensedeep/dynamodb-onetable#286] [Q&A] [answered]
With an entity defined as follows:
```
    Team: {
      pk: { type: String, value: '${_type}#${id}' },
      sk: { type: String, value: '${_type}' },
      gsi1pk: { type: String, value: '${_type}' },
      gsi1sk: { type: String, value: '${_type}#${id}' },
```
I'm trying to figure out how to retrieve all Teams back.  The gsi1 index was set up for that and a query AWS for 'Team' returns all teams.  But I can't figure out the proper call using OneTable.  Neither of these work:
```
        items = await DataDefs.TeamEntity.find(undefined, { index: 'gsi1' });
        items = await DataDefs.TeamEntity.find({}, { index: 'gsi1' });
```
I'm using v2.3.2.

**Comment by mobsense** [comment, marked as answer]
Your second one should work. 

Can you add "log: true" to your params and show the actual command used.

Also, how are you defining the TeamEntity value?

**Comment by jduteau** [comment]
Turning on the logs (which should have been the first thing I did!) showed me that the query was right.  When I went and looked at the data, I realized that my hand-added items had gsi1SK instead of gsi1sk.  User error.
